ironfish-rust/src/note.rs

```diff
@@ -196,11 +190,10 @@ impl<'a> Note {
     /// actually read the contents.
     pub fn encrypt(&self, shared_secret: &[u8; 32]) -> [u8; ENCRYPTED_NOTE_SIZE + aead::MAC_SIZE] {
         let mut bytes_to_encrypt = [0; ENCRYPTED_NOTE_SIZE];
-        bytes_to_encrypt[..11].copy_from_slice(&self.owner.diversifier.0[..]);
-        bytes_to_encrypt[11..43].clone_from_slice(self.randomness.to_repr().as_ref());
+        bytes_to_encrypt[..32].clone_from_slice(self.randomness.to_repr().as_ref());
 
-        LittleEndian::write_u64_into(&[self.value], &mut bytes_to_encrypt[43..51]);
-        bytes_to_encrypt[51..].copy_from_slice(&self.memo.0[..]);
+        LittleEndian::write_u64_into(&[self.value], &mut bytes_to_encrypt[32..40]);
+        bytes_to_encrypt[40..].copy_from_slice(&self.memo.0[..]);
```

Comment:
I'd like to possibly suggest that there be constants representing byte length values to avoid potential proof errors?